lib/Interpreter/Value.cpp

```diff
@@ -231,9 +231,15 @@ namespace cling {
   namespace valuePrinterInternal {
     std::string printTypeInternal(const Value& V);
     std::string printValueInternal(const Value& V);
+    extern const char* const kUndefined;
   } // end namespace valuePrinterInternal
 
   void Value::print(llvm::raw_ostream& Out, bool Escape) const {
+    if (!m_Interpreter) {
+      Out << valuePrinterInternal::kUndefined << "\n";
```

Comment:
Thanks! Do you agree that this is awkward:
```
 V // CHECK: (cling::Value &) <<<invalid>>> @0x{{.*}}
  		  
 +V.dump(); // CHECK-NEXT: <undefined>
```

For all but this case, `V.dum()` is the value part of the value printer. Can we make this:
```
 V // CHECK: (cling::Value &) <<<invalid>>> @0x{{.*}}
  		  
 +V.dump(); // CHECK-NEXT: <<<invalid>>>
```

Regarding
> can the `return "ERROR in cling's callPrintValue(): missing value string.";` be put back to `<undefined>`

That's fine with me. With that change, `<undefined>` would be printed whenever there is an infrastructure issue - that's perfect. Thank you!